opm/material/fluidsystems/blackoilpvt/Co2GasPvt.cpp

```diff
@@ -52,16 +52,44 @@ initFromState(const EclipseState& eclState, const Schedule&)
     size_t numRegions = 1;
     setNumRegions(numRegions);
     size_t regionIdx = 0;
-    Scalar T_ref = eclState.getTableManager().stCond().temperature;
-    Scalar P_ref = eclState.getTableManager().stCond().pressure;
 
-    // Throw an error if STCOND is not (T, p) = (15.56 C, 1 atm) = (288.71 K, 1.01325e5 Pa)
-    if (T_ref != Scalar(288.71) || P_ref != Scalar(1.01325e5)) {
-        OPM_THROW(std::runtime_error, "CO2STORE/CO2SOL can only be used with default values for STCOND!");
+    // Currently we only support constant salinity
+    const Scalar molality = eclState.getTableManager().salinity(); // mol/kg
+    const Scalar MmNaCl = 58.44e-3; // molar mass of NaCl [kg/mol]
+    // convert to mass fraction
+    salinity_[regionIdx] = 1 / ( 1 + 1 / (molality*MmNaCl));
+
```

Comment:
This sets the salinity used by the brine module to compute the fraction of vaporized water in the gas phase. This makes the brine module consistent with the CO2 module. Test is also the reason for the test failure.